docs/paper/dev/getting-started/project-setup.mdx

```diff
@@ -187,6 +187,93 @@ If everything went well, you should see something like this:
 
 ![](assets/paper-plugin-overview.png)
 
+## Plugin remapping
+
+As of 1.20.5, Paper ships with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings.
+If you are using Spigot/Bukkit plugins, your plugin will be assumed to be Spigot-mapped.
+This means that the server will have to deobfuscate and remap the plugin JAR when it's loaded for the first time.
+
+:::info
+
+`paperweight-userdev` already sets this attribute automatically. For more information see the [userdev](./userdev) documentation.
+
+:::
+
+### Mojang mappings
+
+To tell the server that your plugin is Mojang-mapped, you need to add the following code to your build script:
+
+:::note[Paper plugins]
+
+If you are using Paper plugins, this step is not needed as plugins will be assumed to be Mojang-mapped.
+
+:::
+
+```kotlin title="build.gradle.kts"
+tasks.jar {
+    manifest {
+        attributes["paperweight-mappings-namespace"] = "mojang"
+    }
+}
+
+// if you have shadowJar configured
+tasks.shadowJar {
+    manifest {
+        attributes["paperweight-mappings-namespace"] = "mojang"
+    }
+}
+```
+
+```xml title="pom.xml"
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
+    <artifactId>maven-jar-plugin</artifactId>
+    <version>3.4.1</version>
+    <configuration>
+        <archive>
+            <manifestEntries>
+                <paperweight-mappings-namespace>mojang</paperweight-mappings-namespace>
+            </manifestEntries>
+        </archive>
+    </configuration>
+</plugin>
+```
```

Comment:
Can you put these in a tab pane? See Velocity creating-your-first-plugin for info

Reply:
👍🏻

Reply:
done